src/file_cache.rs

```diff
@@ -196,4 +198,88 @@ impl FileCache {
 
         Err(format!("file not found ‘{}’", filename))
     }
+
```

Comment:
This should not longer be necessary, this is now in https://github.com/hyperledger-labs/solang/blob/main/src/sema/file.rs

`formated_message()` no longer needs a file cache argument in the latest version because of it.

Reply:
I print the complete line when I am printing `Diagnotics`. I needed to fetch the line from somewhere, so I added `FileCache` again to formatted message. The function in line 203 retrieves the line and the offsets so that I can print an indicator (^) to the point in the line where there is an error. Can I do the same using only `File`?

Reply:
`get_line_and_offset_from_loc()` belongs in `File` and `struct FileOffsets` should be removed.

Reply:
I updated the comment for this function. Please, take a look now. I can also change the name to make it clearer that the function retrieves a line from the source code.